scylla/src/transport/load_balancing/default.rs

```diff
@@ -2582,5 +2594,29 @@ mod latency_awareness {
                 .await;
             }
         }
+
+        // This is a regression test for #696.
+        #[tokio::test]
+        #[ntest::timeout(1000)]
+        async fn latency_aware_query_completes() {
+            let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+
+            let policy = DefaultPolicy::builder()
+                .latency_awareness(LatencyAwarenessBuilder::default())
+                .build();
+            let handle = ExecutionProfile::builder()
+                .load_balancing_policy(policy)
+                .build()
+                .into_handle();
+
+            let session = SessionBuilder::new()
+                .known_node(uri)
+                .default_execution_profile_handle(handle)
+                .build()
+                .await
+                .unwrap();
+
+            session.query("whatever", ()).await.unwrap_err();
```

Comment:
This statement should send a valid query and check that it succeeded. If it starts failing due to some unrelated reason the test will stop being useful.

Reply:
This test was meant as a regression test, so it only tests that the test completes. We could write some other sophisticated integration tests to ensure that LAP does not cause unpredictable behaviour.